src/x86/nt.rs

```diff
@@ -0,0 +1,275 @@
+pub fn _mm_stream_sd(addr: &mut NonTemporalStoreable<'_, f64>, v: __m128d) {
+    unsafe { arch::_mm_stream_sd(addr.inner.as_ptr(), v) }
+}
+
+/// Store a 32-bit value into a memory location. To minimize caching, the data is flagged as
+/// non-temporal (unlikely to be used again soon).
+#[inline]
+#[target_feature(enable = "sse2")]
+pub fn _mm_stream_si32(addr: &mut NonTemporalStoreable<'_, i32>, v: i32) {
+    unsafe { arch::_mm_stream_si32(addr.inner.as_ptr(), v) }
+}
+
+/// Store a 32-bit value into a memory location. To minimize caching, the data is flagged as
+/// non-temporal (unlikely to be used again soon).
+#[inline]
+#[target_feature(enable = "sse2")]
+pub fn _mm_stream_si128(addr: &mut NonTemporalStoreable<'_, __m128i>, v: __m128i) {
+    unsafe { arch::_mm_stream_si128(addr.inner.as_ptr(), v) }
+}
+
+/// Store a 32-bit part `v.0` of a 128-bit vector into a memory location. To minimize caching, the
+/// data is flagged as non-temporal (unlikely to be used again soon).
+#[inline]
+#[cfg(any())]
+#[target_feature(enable = "sse4a")]
+pub fn _mm_stream_ss(addr: &mut NonTemporalStoreable<'_, f32>, v: __m128) {
+    unsafe { arch::_mm_stream_ss(addr.inner.as_ptr(), v) }
+}
+
+/// Store the four double precision floats of a 256-bit vector into a memory location. To minimize
+/// caching, the data is flagged as non-temporal (unlikely to be used again soon).
+#[inline]
+#[target_feature(enable = "avx")]
+pub fn _mm256_stream_pd(addr: &mut NonTemporalStoreable<'_, __m256d>, v: __m256d) {
+    unsafe { arch::_mm256_stream_pd(addr.inner.as_ptr() as *mut f64, v) }
+}
+
+/// Store eight single precision float values of a 256-bit vector into a memory location. To
+/// minimize caching, the data is flagged as non-temporal (unlikely to be used again soon).
+#[inline]
+#[target_feature(enable = "avx")]
+pub fn _mm256_stream_ps(addr: &mut NonTemporalStoreable<'_, __m256>, v: __m256) {
+    unsafe { arch::_mm256_stream_ps(addr.inner.as_ptr() as *mut f32, v) }
+}
+
+/// Store a 256-bit vector into an aligned memory location. To minimize caching, the data is
+/// flagged as non-temporal (unlikely to be used again soon).
+#[inline]
+#[target_feature(enable = "avx")]
+pub fn _mm256_stream_si256(addr: &mut NonTemporalStoreable<'_, __m256i>, v: __m256i) {
+    unsafe { arch::_mm256_stream_si256(addr.inner.as_ptr(), v) }
+}
+
+/// A pointer to non-temporally written-to memory.
+///
+/// The lifetime on this struct means: we can write to the memory within lifetime `'data` while
+/// guaranteeing an appropriate fence afterwards, and then that effect appears as if we had written
+/// to the data through a mutable reference. The role of this type in this is the guarantee that
+/// *no* active reference, mutable or shared, to the memory can exist in the lifetime `'data' which
+/// could observe the memory while it is in the non-coherent state between having been written the
+/// points in time where it is written-to non-temporally and the fence being issue.
```

Comment:
Grammar nit, don't think I've changed the meaning of what you've said but feel free to modify if so.
```rust
/// The lifetime on this struct means that we can write to the memory within
/// lifetime `'data` while guaranteeing an appropriate fence afterwards, and
/// then that effect appears as if we had written to the data through a mutable
/// reference.

/// The role of this type is to guarantee that *no* active reference to the
/// memory, mutable or shared, can exist during the lifetime `'data' which could
/// observe the memory while it is in the non-coherent state between having been
/// written to non-temporally and the fence being issued.
```